tket/src/Circuit/ThreeQubitConversion.cpp

```diff
@@ -127,30 +151,44 @@ static std::pair<Circuit, Complex> two_qubit_plex(
   }
   // 3. Compute R such that U0 = L D R and U1 = L D* R.
   Eigen::Matrix4cd R = D * L.adjoint() * U1;
-  // 4. Decompose R into a 2-CX circuit followed by a diagonal.
-  auto [R_circ, w0] = decompose_2cx_DV(R);
-  Complex w1 = std::conj(w0);
-  // 5. Construct the circuit.
-  // The diagonal from R's decomposition commutes forward through the controls
-  // on qubits 1 and 2, so can be absorbed into L.
-  Circuit circ(3);
-  unit_map_t qm;
-  qm.insert({Qubit(0), Qubit(1)});
-  qm.insert({Qubit(1), Qubit(2)});
-  circ.append_with_map(R_circ, qm);
-  circ.append(two_qubit_diag_adjoint_plex(D));
-  L.col(0) *= w0;
-  L.col(1) *= w1;
-  L.col(2) *= w1;
-  L.col(3) *= w0;
-  if (extract_final_diagonal) {
-    auto [L_circ, z0] = decompose_2cx_DV(L);
+
+  // We try conjugating the L and R circuits to see if we can reduce CX count.
+  std::optional<Circuit> best_circ;
+  Complex best_z0 = 0;
+  unsigned best_n_cx = 0;
```

Comment:
Shouldn't this start high?

Reply:
It doesn't matter -- I just added it because of compiler warnings. But I can set `best_n_cx` to max unsigned if you prefer :)

Reply:
OK my mistake, it's fine.